File: src/unity/Runtime/FacebookAds/IFacebookAds.cs
using System.Threading.Tasks;

namespace EE {
    public interface IFacebookAds : IPlugin {
        Task<bool> Initialize();
        string GetTestDeviceHash();
        void AddTestDevice(string hash);
        void ClearTestDevices();
        IBannerAd CreateBannerAd(string adId, FacebookBannerAdSize adSize);
        IFullScreenAd CreateInterstitialAd(string ad);
        IFullScreenAd CreateRewardedAd(string adId);
    }
}

File: src/ts/src/facebook_ads/IFacebookAds.ts
import {
    IBannerAd,
    IFullScreenAd,
} from "../ads";
import { IPlugin } from "../core";
import { FacebookBannerAdSize } from "./FacebookBannerAdSize";

export interface IFacebookAds extends IPlugin {
    initialize(): Promise<boolean>;
    getTestDeviceHash(): string;
    addTestDevice(hash: string): void;
    clearTestDevices(): void;
    createBannerAd(adId: string, adSize: FacebookBannerAdSize): IBannerAd;
    createInterstitialAd(adId: string): IFullScreenAd;
    createRewardedAd(adId: string): IFullScreenAd;
}